code/datums/components/bump_attack.dm

```diff
@@ -0,0 +1,57 @@
+/datum/component/bump_attack
+    var/active = FALSE
+    var/datum/action/bump_attack_toggle/toggle_action
+
+/datum/component/bump_attack/Initialize()
+	. = ..()
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(isliving(parent))
+		toggle_action = new()
+		RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, .proc/bump_attack_toggle)
+		toggle_action.give_action(parent)
+		toggle_action.update_button_icon(active)
+	else
+		return COMPONENT_INCOMPATIBLE
```

Comment:
resolve this earlier with an early return.

Reply:
This was done this way to account for the fact that this will be used by other movable atoms, such as vehicles.
If we wanted it only for living we could do instead `if(!isliving)` instead of `if(!ismovableatom)`
Expansibility and all.

Reply:
Logic changed now.